### ISSUE TITLE
After upgrade 2.4 to 2.5 can not update permisions in project

### ISSUE THREAD

**Issue by homiakos** [rancher/rancher#30030]
**What kind of request is this question/bug:**
I have following configuration:
1. Separate cluster k8s for rancher  v1.18.5 
rancher 2.4.5
Other k8s cluster for management

After upgrade i can not upgade permision in project
i get the following error : 
`Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: Service Unavailable`

we use a proxy and I tried to change the proxy settings:

first try:
`noProxy: "127.0.0.1,10.100.8.32,10.100.24.116,localhost,10.233.0.1,169.254.169.254,10.43.0.0/16,10.233.0.0/16,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.humans.dc,.local,.ftm.humans.uz,.humans.dev,.cattle-system.svc,.svc,.cluster.local"
`

and second try(default values):
` 127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local`

**both don't work**

AFTER i tried delete deployment rancher-webhooks and redeploy chart rancher, but no new deploy appeared. as I understand it, this deployment has been placed on a separate chart:

https://github.com/rancher/webhook/tree/master/charts/rancher-webhook

How redeploy this chart correctly?

How fix this?
Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: Service Unavailable

**Environment information**
- Rancher version chart 2.4.5 upgrade 2.5.2
- Installation option HA

**Cluster information**
- Kubernetes version (use `kubectl version`): v1.18.5

**Comment by homiakos** [comment]
after 20 minutes upgrade rancher chart there was a chart webhook-charts, a deployment and pod.
But you still can't add or remove project members
Now i have error:
Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: Service Unavailable

from pod 

`root@rancher-77b46946cb-bxdlt:/var/lib/rancher# printenv |grep PROXY
HTTP_PROXY=http://proxy
NO_PROXY=127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.svc,.cluster.local
HTTPS_PROXY=http://proxy
`

`root@rancher-77b46946cb-bxdlt:/var/lib/rancher# curl -vvv https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s -k
*   Trying 10.233.34.253...
* TCP_NODELAY set
* Connected to rancher-webhook.cattle-system.svc (10.233.34.253) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Client hello (1):
* TLSv1.3 (OUT), TLS Unknown, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: O=dynamic; CN=dynamic
*  start date: Nov 11 10:08:46 2020 GMT
*  expire date: Nov 11 11:27:41 2021 GMT
*  issuer: O=dynamiclistener-org; CN=dynamiclistener-ca
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* Using Stream ID: 1 (easy handle 0x560220f86580)
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
> GET /v1/webhook/validation?timeout=10s HTTP/2
> Host: rancher-webhook.cattle-system.svc
> User-Agent: curl/7.58.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
< HTTP/2 500 
< content-type: text/plain; charset=utf-8
< x-content-type-options: nosniff
< content-length: 4
< date: Wed, 11 Nov 2020 13:44:12 GMT
< 
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
EOF
* Connection #0 to host rancher-webhook.cattle-system.svc left intact
`

**Comment by jiaqiluo** [comment]
Hi @homiakos I am trying to reproduce the bug and need to collect more information form you:

From what you describe, you have Rancher `v2.4.5` running on a k8s `v1.18.5` cluster, and Rancher is configured to use a proxy when installing it. you upgrade Rancher from `v2.4.5` to `v2.5.2`

> After upgrade I can not upgrade permission in project

Questions: 
1. Which project do you mention here? It the system project of the local cluster, or a project in other Rancher-manged downstream clusters? 
2. Also, what do you mean by `upgrading permission`?

**Comment by homiakos** [comment]
Hi, @jiaqiluo 

> 1. Which project do you mention here? Is it the system project of the local cluster, or a project in other Rancher-manged downstream clusters?

Which project do you mention here? Is it the system project of the local cluster, or a project in other Rancher-manged downstream clusters?
The problem is reproducible when trying to Add / Remove a Project member from both the local cluster and the remote one.
I repeat the problem with access to rancher-webhook.cattle-system.svc:443


> 2. Also, what do you mean by `upgrading permission`?

 Add / Remove a Project member

**Comment by mfbrj** [comment]
Similar problem here. And I'm not behind a proxy (I was behind a proxy and I removed the proxy to try to solve the problem and unfortunally the problem remains).

I'd upgraded Rancher from 2.4.8 to 2.5.5 (in a k3s local cluster) and now **I can not**:

- Remove users from `https://myrancherdomain.com/g/security/accounts/users`
- Add or edit Roles such as in `https://myrancherdomain.com/g/security/roles/add?context=project&id=rt-8mfft` or `https://myrancherdomain.com/g/security/roles/edit/rt-8mfft?type=project`

Error in Rancher UI:
```
Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: Service Unavailable
```
Recurring error in Rancher console logs:
```
2021/02/01 14:25:31 [INFO] [mgmt-auth-users-controller] Deleting globalRoleBinding grb-ll2wl for user u-jvkwxwsir2
2021/02/01 14:25:31 [ERROR] error syncing 'u-jvkwxwsir2': handler mgmt-auth-users-controller: error deleting global role template grb-ll2wl: Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: Service Unavailable, requeuing
```

Rancher yaml used to install:
```
additionalTrustedCAs: true
replicas: 2
auditLog:
  level: 3
hostname: myrancherdomain.com
ingress:
  tls:
    source: secret
privateCA: true
```

```
$ helm version
version.BuildInfo{Version:"v3.3.1", GitCommit:"249e5215cde0c3fa72e27eb7a30e8d55c9696144", GitTreeState:"clean", GoVersion:"go1.14.7"}
```
Rancher version: 2.5.5
K3s version (local cluster): v1.18.9+k3s1

**Comment by mfbrj** [comment]
Additional information:

```
root@rancher-758bd88dd7-fgdrq:/var/lib/rancher# curl -vvv https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s -k
*   Trying 10.43.201.151...
* TCP_NODELAY set
* Connected to rancher-webhook.cattle-system.svc (10.43.201.151) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Client hello (1):
* TLSv1.3 (OUT), TLS Unknown, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: O=dynamic; CN=dynamic
*  start date: Jan  8 17:10:15 2021 GMT
*  expire date: Jan  8 17:10:15 2022 GMT
*  issuer: O=dynamiclistener-org; CN=dynamiclistener-ca
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* Using Stream ID: 1 (easy handle 0x555e4304f5c0)
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
> GET /v1/webhook/validation?timeout=10s HTTP/2
> Host: rancher-webhook.cattle-system.svc
> User-Agent: curl/7.58.0
> Accept: */*
> 
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
* TLSv1.3 (OUT), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
< HTTP/2 500 
< content-type: text/plain; charset=utf-8
< x-content-type-options: nosniff
< content-length: 4
< date: Mon, 01 Feb 2021 17:19:28 GMT
< 
* TLSv1.3 (IN), TLS Unknown, Unknown (23):
EOF
* Connection #0 to host rancher-webhook.cattle-system.svc left intact
```

**Comment by mfbrj** [comment]
Solved!

Solution: Update k3s where Rancher is installed to `v1.20.2+k3s1`

**Comment by mfbrj** [comment]
Despite the upgrade in k3s to `v1.20.2+k3s1` that I manually did (via command line) on both of my nodes solves the `rancher-webhook` error, I figure out 2 problems:

1. Rancher UI was still displaying the previous `v1.18.8+k3s1` k3s version on the `Edit cluster` screen.
2. All charts are gone from the `Deploy Chart` (https://myrancherdomain.com/dashboard/c/local/apps/charts) screen in Rancher UI in the `local` cluster.

So, I [downgrade k3s](https://medium.com/@madushagunasekara/downgrade-k3s-version-b8cf3fdbb718) to `v1.18.8+k3s1` on both nodes.

After that, I did a k3s upgrade (this time using the **Rancher UI**) in `local` cluster to the last version available (`v1.19.7+k3s1`).

All works fine, with no rancher-webhook errors and all charts displaying again in `Deploy Chart` screen.

**Comment by mbriquemont** [comment]
Hello everyone,
I have a similar issue since I upgraded my Rancher setup from 2.4.8 -> 2.5.7.

The current setup is a pair of 2 VMs running K3OS and a K3S cluster on which Rancher is deployed.

I upgraded to the latest K3OS version 0.11.1 and K3S cluster is in v1.18.9+k3s1 but the issue still remains.

As the VMs are behind a proxy, here is the Rancher config I used to deploy in with Helm:
```
hostname: ranchertest.mydomain.com
ingress:
  tls:
    source: secret
noProxy: .mylocalcompanydomain.com,localhost,127.0.0.0/8,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,cattle-system.svc,.svc,.cluster.local
proxy: http://mylocalcompanydomain.com:8080
```

Following the upgrade, I tried to add a project and to a user application cluster suring the Rancher GUI and got the following error :
```
Failed while: Wait for Condition: BackingNamespaceCreated: True
```

The logs of the rancher pods gave me this :
```
[pod/rancher-6868f6d7fd-xhzf7/rancher] 2021/03/18 18:39:08 [INFO] Creating globalRoleBindings for u-3uipxci44p
[pod/rancher-6868f6d7fd-xhzf7/rancher] 2021/03/18 18:39:08 [ERROR] error syncing 'c-76dgz/p-jwlxc': handler pipeline-controller: Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: x509: certificate signed by unknown authority, requeuing
```

by kubectl exec -it in the rancher pod i tried these commands:
```
root@rancher-6868f6d7fd-xhzf7:/var/lib/rancher# openssl s_client -connect rancher-webhook.cattle-system.svc:443
CONNECTED(00000005)
depth=0 O = dynamic, CN = dynamic
verify error:num=20:unable to get local issuer certificate
verify return:1
depth=0 O = dynamic, CN = dynamic
verify error:num=21:unable to verify the first certificate
verify return:1
---
Certificate chain
 0 s:O = dynamic, CN = dynamic
   i:O = dynamiclistener-org, CN = dynamiclistener-ca
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIBqTCCAVCgAwIBAgIIW/BLn7n/zkwwCgYIKoZIzj0EAwIwOzEcMBoGA1UEChMT
ZHluYW1pY2xpc3RlbmVyLW9yZzEbMBkGA1UEAxMSZHluYW1pY2xpc3RlbmVyLWNh
MB4XDTIxMDMxNjA3MjE1NFoXDTIyMDMxNjA3MjE1NFowJDEQMA4GA1UEChMHZHlu
YW1pYzEQMA4GA1UEAxMHZHluYW1pYzBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IA
BPhKI9KsrcYstuM++Tzs5WH/mT3nQerM5jt4I7DultXa6NM8e6+6SD2hA7hF0At2
dnGTPK4PeBx5u10KEQOcStKjVTBTMA4GA1UdDwEB/wQEAwIFoDATBgNVHSUEDDAK
BggrBgEFBQcDATAsBgNVHREEJTAjgiFyYW5jaGVyLXdlYmhvb2suY2F0dGxlLXN5
c3RlbS5zdmMwCgYIKoZIzj0EAwIDRwAwRAIgCGWOOBuTcFC2o0IslLsX2OaLb2Gx
Zq6vZRshMJEY0YcCIGPsbCkt9+wzc5jP1VtCclqOThBBT89WV8E2IbCT7orJ
-----END CERTIFICATE-----
subject=O = dynamic, CN = dynamic

issuer=O = dynamiclistener-org, CN = dynamiclistener-ca

---
No client certificate CA names sent
Peer signing digest: SHA256
Peer signature type: ECDSA
Server Temp Key: X25519, 253 bits
---
SSL handshake has read 800 bytes and written 415 bytes
Verification error: unable to verify the first certificate
---
New, TLSv1.3, Cipher is TLS_AES_256_GCM_SHA384
Server public key is 256 bit
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
Early data was not sent
Verify return code: 21 (unable to verify the first certificate)
---
---
Post-Handshake New Session Ticket arrived:
SSL-Session:
    Protocol  : TLSv1.3
    Cipher    : TLS_AES_256_GCM_SHA384
    Session-ID: A112E730A9E877B359595BBC6541E65268507F8B208B8A249B73760216707A14
    Session-ID-ctx:
    Resumption PSK: AC658D293378E9E065AE4E75800948CFFC2FDB2080BD821E7D9EEDA4CC45948BE98E684C560A3CA961D98FC42E8765D5
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 604800 (seconds)
    TLS session ticket:
    0000 - 4a f4 88 1c 93 ba bc 7d-cc bb 28 d7 65 ec 53 c3   J......}..(.e.S.
    0010 - c8 9f 8f af a9 19 21 e1-83 c9 ac 72 36 b0 84 4b   ......!....r6..K
    0020 - a5 fc e2 e2 67 15 cf 93-42 17 85 81 77 fd 5f a4   ....g...B...w._.
    0030 - 47 72 40 64 8e 55 ea 31-2a 42 7c a9 52 94 38 01   Gr@d.U.1*B|.R.8.
    0040 - e0 b7 50 33 31 55 95 64-f5 f0 db c8 6c 3d 2c 7a   ..P31U.d....l=,z
    0050 - a8 d8 82 cc f7 fc 45 f6-5c cb e0 54 f4 b1 da d9   ......E.\..T....
    0060 - 1b cc 58 f1 d0 02 74 32-72 bf e8 39 f0 e5 99 fb   ..X...t2r..9....
    0070 - 4f be 55 fe 28 6d b3 40-ff a5 a7 e6 40 b9 af 54   O.U.(m.@....@..T
    0080 - 56                                                V

    Start Time: 1616093032
    Timeout   : 7200 (sec)
    Verify return code: 21 (unable to verify the first certificate)
    Extended master secret: no
    Max Early Data: 0
---
read R BLOCK
```

and 
```
root@rancher-6868f6d7fd-xhzf7:/var/lib/rancher# curl -vvv https://rancher-webhook.cattle-system.svc:443/v1/webhook
*   Trying 10.43.128.8...
* TCP_NODELAY set
* Connected to rancher-webhook.cattle-system.svc (10.43.128.8) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/rancher/ssl
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Unknown (8):
* TLSv1.3 (IN), TLS Unknown, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (OUT), TLS alert, Server hello (2):
* SSL certificate problem: unable to get local issuer certificate
* stopped the pause stream!
* Closing connection 0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

It seems the certificate of the rancher-webhook is not trusted from withing the rancher pod.  Any ideas on what could be the problem here ?

Thanks in advance

Mathieu

**Comment by kgk** [comment]
Has this been resolved? I installed rancher 2.5.7 using helm onto a eks hosted cluster and am seeing similar errors while trying to create seperate EKS cluster.   
![Screen Shot 2021-04-21 at 10 51 41 PM](https://user-images.githubusercontent.com/1790880/115662730-3fbcf480-a2f4-11eb-8967-ef0226f5502b.jpg)

**Comment by mbriquemont** [comment]
Hi, I ended up upgrading K3S to  v1.20.4+k3s1, as mentioned by @mfbrj and it fixed the issue for me.


I would really like to know how this upgrade fixed the issue though. 

Before upgrading K3S, I tested adding the "issuer=O = dynamiclistener-org, CN = dynamiclistener-ca" certificate to the OS trusted CA list as it was missing.  It fixed the SSL issue that I had with curl but not the one in rancher unfortunately.
